jlox/app/src/main/java/dev/zxul767/lox/runtime/LoxList.java

```diff
@@ -3,62 +3,100 @@
 import dev.zxul767.lox.parsing.Token;
 import dev.zxul767.lox.parsing.TokenType;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.function.Function;
 
-@FunctionalInterface
-interface Invoker<T, R> {
-  R invoke(LoxListInstance self, T[] args);
-}
-
 class LoxList extends LoxClass {
-  // constructor call
-  @Override
-  public Object call(Interpreter interpreter, List<Object> args) {
-    return new LoxListInstance(this);
+  public static final Map<String, CallableSignature> signatures;
+  static {
+    List<CallableSignature> members = Arrays.asList(
+        new CallableSignature("__init__", "list"),
+        new CallableSignature("length", "number"),
+        new CallableSignature("append", new Parameter("value"), "bool"),
+        new CallableSignature("at", new Parameter("index", "int")),
+        new CallableSignature("__getitem__", new Parameter("index", "int")),
+        new CallableSignature(
+            "set", Parameter.list("index", "int").add("value").build()
+        ),
+        new CallableSignature(
+            "__setitem__", Parameter.list("index", "int").add("value").build()
+        ),
+        new CallableSignature("clear", "nil"), new CallableSignature("pop")
+    );
+    signatures = new HashMap<>();
+    for (CallableSignature signature : members) {
+      signatures.put(signature.name, signature);
+    }
+  }
+  static Map<String, LoxCallable> buildConstructorMethod() {
```

Comment:
How about `createPredefinedMethods`?

Reply:
Ended up being `createDunderMethods`. Not as obvious as `createPredefinedMethods`, but more expressive about our real intent here (at least if you're familiar with Python)